ISSUE-LABELLING.md

```diff
@@ -0,0 +1,233 @@
+* comp/secio
+* comp/tcp
+* comp/tls
+* comp/webrtc
+* comp/wss
+* comp/yamux
+
+### Difficulty (`dif/`)
+
+> Subjectively estimates the difficulty of resolution of an issue, based on
+> prior expertise about libp2p and its constructs, p2p networking, systems
+> programming, etc.
+
+Values and calibration guidance:
+
+* dif/trivial: Can be confidently tackled by newcomers, who are widely
+  unfamiliar with libp2p.
+* dif/easy: A prexisting libp2p user should be able to pick this up.
+* dif/medium: Prior experience in having developed libp2p modules is likely
+  helpful.
+* dif/hard: Having worked 
+* dif/expert: Requires extensive knowledge of the history, implications,
+  ramifications of the issue, as well as deep understanding of the libp2p stack
+  and experience with systems and network programming.
+
+### Size (`size/`)
+
+> Estimates the amount of work. This axis is distinct to the complexity, e.g.
+> something can be easy but require a lot of work, or vice versa. Use 
+
+Values and calibration guidance:
+
+* size/XS: minutes.
+* size/S: several hours.
+* size/M: one day.
+* size/L: 2-3 days.
+* size/XL: 1 week.
+* size/XXL: > 1 week.
+
+### Kind (`kind/`)
+
+> Describes the nature of the issue.
+
+* kind/bug
+* kind/improvement
+* kind/tracking-issue
+* kind/question
+* kind/enhancement
+* kind/architecture
+* kind/discussion
+
+### Impact (`impact/`)
+
+> Contextualises the impact, and therefore the perceived/derived severity, of
+> the issue.
+
+* impact/regression
+* impact/api-breakage
+* impact/quality
+* impact/dx
+* impact/test-flakiness
+
+### Topic (`topic/`)
+
+> Classifies issues pertaining to particular themes. Topics can be created to
+> refer to epics, larger bodies of work, or ad-hoc endeavours.
+
+* topic/interoperability
+* topic/docs
+* topic/infra
+
+## Execution labels
+
+These labels describe some aspect of the execution on the issue. These labels
+are dynamic; they change as a result of planning, analysis, scoping, and
+generally applying the development workflow. They are usually set by
+maintainers.
+
+### Priority (`Pn`)
+
+> Indicates the current implementation priority; used for planning and
+> roadmapping purposes.
+
+* P0: Critical.
+* P1: High.
+* P2: Medium.
+* P3: Low.
+
+### Hints/advisories (`hint/`)
+
+> Surfaces observations and facts about issues, facilitating progress to be
+> made.
+
+* hint/good-first-issue
+* hint/needs-contributor
+* hint/needs-participation
+* hint/needs-decision
+* hint/needs-triage
+* hint/needs-analysis
+* hint/needs-author-input
+* hint/needs-team-input
+* hint/needs-community-input
+* hint/needs-review
+* hint/needs-help
+
+### Status (`status/`)
+
+> Workflow-relevant. Marks the current status of the issue. 
+
+* status/done
+* status/deferred
+* status/in-progress
+* status/blocked
+* status/inactive
+* status/waiting
+* status/rotten
+
+## Implementation plan
+
+1. Create labels on go-libp2p
+   [declaratively](https://github.com/micnncim/action-label-syncer).
```

Comment:
We made a tool that we've used multiple times to sync the Priority and Difficulty labels in the past, check https://github.com/ipfs/team-mgmt/tree/master/tools/sync-labels

Reply:
We have GitHub actions now, but thanks for the pointer! ;-)